src/Workspaces/Core/MSBuild.BuildHost/Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost.csproj

```diff
@@ -21,16 +21,28 @@
          -->
     <PlatformTarget>AnyCPU</PlatformTarget>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+
+    <!--
+      We specifically reference Microsoft.Build 17.3.4, since it's the oldest version still compatible $(NetRoslynBuildHostNetCoreVersion).
+      The version targeted is only used for build time, since we use MSBuildLocator to discover the proper version at runtime.
+    -->
+    <_MsbuildVersion>17.3.4</_MsbuildVersion>
   </PropertyGroup>
+  <ItemGroup Condition="'$(DotNetBuildSourceOnly)' == 'true'">
```

Comment:
There's no reason why you need `PackageReference` for sourcebuild.  It should work just the same there.

Reply:
I see what happened, in SBRP the `net472` asset is missing: https://github.com/dotnet/source-build-reference-packages/tree/main/src/referencePackages/src/microsoft.build.framework/17.3.4/ref

Reply:
Would you mind filing an issue? We can remove the condition here once fixed.